app/src/main/java/org/oppia/app/options/OptionsFragmentPresenter.kt

```diff
@@ -133,31 +133,65 @@ class OptionsFragmentPresenter @Inject constructor(
   }
 
   private enum class ViewType {
-    VIEW_TYPE_STORY_TEXT_SIZE,
+    VIEW_TYPE_READING_TEXT_SIZE,
     VIEW_TYPE_APP_LANGUAGE,
     VIEW_TYPE_AUDIO_LANGUAGE
   }
 
-  fun updateStoryTextSize(textSize: String) {
+  fun updateReadingTextSize(textSize: String) {
     when (textSize) {
-      getOptionControlsItemViewModel().getStoryTextSize(StoryTextSize.SMALL_TEXT_SIZE) -> {
-        profileManagementController.updateStoryTextSize(profileId, StoryTextSize.SMALL_TEXT_SIZE)
-        storyTextSize = StoryTextSize.SMALL_TEXT_SIZE
+      getOptionControlsItemViewModel().getReadingTextSize(ReadingTextSize.SMALL_TEXT_SIZE) -> {
+        profileManagementController.updateReadingTextSize(
+          profileId,
+          ReadingTextSize.SMALL_TEXT_SIZE
+        ).observe(
+          fragment,
+          Observer {
+            if (it.isSuccess()) {
```

Comment:
I suggest also logging here & elsewhere when it fails since that can help with investigations in situations like the one you were in before with the failing tests.

Reply:
Done

Reply:
Resolving this on behalf of Ben and also to unblock Mohamed.

Reply:
Resolving this to unblock Mohamed.